pyproject.toml

```
@@ -87,6 +91,9 @@ Examples = "https://github.com/scikit-build/scikit-build-core/tree/main/tests/pa
 cmake_extensions = "scikit_build_core.setuptools.extension:cmake_extensions"
 cmake_source_dir = "scikit_build_core.setuptools.extension:cmake_source_dir"
 
+[project.entry-points.skbuild]
```

```suggestion
[project.entry-points.skbuild]
```

I think this should probably have one more level, so that we can use this space for other things too. Something like

```suggestion
[project.entry-points."scikit-build.metadata"]
```

That leaves room for other entries here later, like extra cmake modules. And it's more symmetric with the metadata field.

What was the problem with a `-`? Ideally, it would be nice to stay consistent if it's possible.

---

Do the users call it like that? Would be a bit confusing having the quotation marks there.

---

Yes, thought "users" in this case would only be people making a new plugin - most users will never see this. And that's part of the spec (nested dicts are not allowed - don't know why), so that's true of anyone making an entry point like this. It's not as bad as poetry's `[tool.poetry.plugins."poetry.application.plugin"]`, though. :D

---

I can do `"scikit_build.metadata"` but dashes are not valid in entrypoint groups (https://packaging.python.org/en/latest/specifications/entry-points/#data-model). Inconvenient that it doesn't match `[tool.scikit-build.metadata]` etc. but, as you say, it will only be used by plugin writers, not by users, so less of a concern.

---

Only underscores in entrypoint groups, and only dashes in entrypoint names. Interesting choice.

We don't own "skbuild" on PyPI, though maybe we should. I'd say `scikit_build` is the technically correct name we should be using for entry points, and thankfully not required by normal users.